packages/blocks/src/store/process-block-type.js

```diff
@@ -33,6 +33,37 @@ const LEGACY_CATEGORY_MAPPING = {
 	layout: 'design',
 };
 
+/**
+ * Merge block variations bootstrapped from the server and client.
+ *
+ * When a variation is registered in both places, its properties are merged.
+ *
+ * @param {Array} bootstrappedVariations - A block type variations from the server.
+ * @param {Array} clientVariations       - A block type variations from the client.
+ * @return {Array} The merged array of block variations.
+ */
+function mergeBlockVariations(
+	bootstrappedVariations = [],
+	clientVariations = []
+) {
+	const result = [ ...bootstrappedVariations ];
+
+	clientVariations.forEach( ( clientVariation ) => {
+		const index = result.findIndex(
+			( bootstrappedVariation ) =>
+				bootstrappedVariation.name === clientVariation.name
```

Comment:
Any concerns with the fact that variations are [not required to have a name](https://developer.wordpress.org/news/2023/08/29/an-introduction-to-block-variations/#variations.js)? Could this possibly run into the issue of wrongly assuming two completely different variations without a name are the same one?

Reply:
Odd. I always thought that a unique name was required; this seems like an oversight on our end.

Reply:
Maybe it's a very rare or legacy scenario, but still worth considering. I was also surprised when I learned about it when reading @ndiego's post linked above.

Maybe we should treat such variations as unique and never attempt to merge them?